src/TDebug.cpp

```diff
@@ -25,22 +25,65 @@
 #include "TConsole.h"
 #include "mudlet.h"
 
+// This is a Unicode NON-character code which is explicitly undisplayable but
+// can be embedded for our own internal purposes:
+const QChar TDebug::csmContinue = QChar(0xFFFF);
+
+QMap<const Host*, QPair<QString, QChar>> TDebug::smIdentifierMap;
+QQueue<QChar> TDebug::smAvailableIdentifiers;
+bool TDebug::initialised = false;
+QQueue<TDebugMessage> TDebug::smMessageQueue;
 
 TDebug::TDebug(const QColor& c, const QColor& d)
-: msg()
-, fgColor(c)
+: fgColor(c)
 , bgColor(d)
 {
 }
 
-// code is a dummy variable so that using operator>>() (at the end of a series
-// of operator<<() calls) causes the accumulated message to be printed/flushed
-// out...
-TDebug& TDebug::operator>>(const int code)
+// This is the method that pushes the accumulated text out to the Central Debug
+// Console. Handles 'msg' beginning with 'csmContinue', otherwise if more than 1
+// profile active, prepends msg with an indicator of the profile from which it
+// came, which is deduced from the supplied Host pointer.
+TDebug& TDebug::operator>>(Host* pHost)
 {
-    Q_UNUSED(code);
-    if (mudlet::mpDebugConsole) {
-        mudlet::mpDebugConsole->print(msg, fgColor, bgColor);
+    if (Q_UNLIKELY(!mudlet::mpDebugConsole)) {
+        if (Q_LIKELY(!msg.isEmpty())) {
+            // Don't enqueue empty messages
+            QChar tag = deduceProfileTag(msg, pHost);
+            TDebugMessage newMessage(msg, tag, fgColor, bgColor);
+            smMessageQueue.enqueue(newMessage);
+        }
+
+    } else {
+        if (Q_UNLIKELY(!smMessageQueue.isEmpty())) {
+            // The mpDebugConsole must have just come on-line - so unload all
+            // the stacked up messages:
+            while (!smMessageQueue.isEmpty()) {
+                const auto& message = smMessageQueue.dequeue();
+                if (message.mTag.isNull()) {
+                    mudlet::mpDebugConsole->print(message.mMessage, message.mForeground, message.mBackground);
+                } else {
+                    mudlet::mpDebugConsole->print(QStringLiteral("[%1] %2").arg(message.mTag, message.mMessage), message.mForeground, message.mBackground);
```

Comment:
Not being able to ***move*** construct the return value because it is generated from a `const` expression (a `QStringLiteral` with parameter replacement) is going to be a pattern that we experience a lot I suspect.